integration/raft/channel_participation_test.go

```diff
@@ -261,9 +261,24 @@ var _ = Describe("ChannelParticipation", func() {
 
 			By("removing orderer1 from a channel")
 			channelparticipation.Remove(network, orderer1, "participation-trophy")
+			Eventually(func() channelparticipation.ChannelInfo {
+				return channelparticipation.ListOne(network, orderer1, "participation-trophy")
+			}, network.EventuallyTimeout).Should(Equal(channelparticipation.ChannelInfo{
+				Name:            "participation-trophy",
+				URL:             "/participation/v1/channels/participation-trophy",
+				Status:          "inactive",
+				ClusterRelation: "none",
+				Height:          0,
+			}))
+
+			By("submitting transaction to orderer1")
+			env := CreateBroadcastEnvelope(network, peer, "participation-trophy", []byte("hello"))
+			resp, err := ordererclient.Broadcast(network, orderer1, env)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(resp.Status).To(Equal(common.Status_BAD_REQUEST))
 
 			By("listing all channels for orderer1")
-			channelparticipation.List(network, orderer1, []string{"another-participation-trophy"})
+			channelparticipation.List(network, orderer1, []string{"participation-trophy", "another-participation-trophy"})
```

Comment:
Is this because the channel is still pending removal? If so this may introduce a race and a flakey test, because if the test thread is slower then the orderer1 process, you won't get to see this transitory state.

Reply:
Yes since the pending removals still exist in the list API now. Need a better way to wait for the removal to complete in the test.

Reply:
Updated the channelparticipation list helper to return all the channels to do a exact comparison